common/export-common-cartridge/src/java/org.sakaiproject.exporter.impl/CCExport.java

```diff
@@ -631,23 +471,36 @@ public String getParent(String s) {
 		}
 	}
 
-	// return relative path to target from base
-	// base is assumed to be "" or ends in /
-	public String relativize(String target, String base) {
-	if (base.equals(""))
-	    return target;
-	if (target.startsWith(base))
-	    return target.substring(base.length());
-	else {
-		// get parent directory of base directory.
-		// base directory ends in /
-		int i = base.lastIndexOf("/", base.length()-2);
-		if (i < 0) {
-			base = "";
-		} else {
-			base = base.substring(0, i + 1); // include /
+	public String removeDotDot(String s) {
+		while (true) {
```

Comment:
I'd consider this style of loops dangerous as if the escape hatch out of the loop isn't perfect it can become an avenue to perform a denial of service as threads get stuck in a loop. If changing the way the loop operates isn't simple then just adding a limit to the loop is a simple way around it so it will only remove up to 1000 "/../" or something like that.